### PR TITLE
Migrate from using AuthFile to using Auth

### LINKED ISSUE #6: Migrate from using AuthFile to using Auth
https://github.com/PlaceOS/google#googleauth

### PR DESCRIPTION
**Pull request by dukenguyenxyz**
- Resolves https://github.com/PlaceOS/wallet_api/issues/6

### CODE REVIEW

File: src/models/auth.cr [hunk 1]
```diff
@@ -1,9 +1,17 @@
 class Auth
+  @signing_key : String
+
   def initialize(@scopes : String)
-    @file_path = File.expand_path(ENV["GOOGLE_AUTH_FILE"])
+    @signing_key = read_base64_key(ENV["GOOGLE_PRIVATE_KEY"])
+    @issuer = ENV["GOOGLE_CLIENT_EMAIL"]
   end
```
Comment (caspiano) [inline src/models/auth.cr]: Looking good, could you throw these under the constants file if there is one.

File: src/config.cr [hunk 2]
```diff
@@ -14,10 +14,10 @@ require "action-controller/server"
 # Configure logging (backend defined in constants.cr)
 if App.running_in_production?
   log_level = Log::Severity::Info
-  ::Log.setup "*", :warning, App::LOG_BACKEND
+  ::Log.setup "*", log_level, App::LOG_BACKEND
 else
   log_level = Log::Severity::Debug
-  ::Log.setup "*", :info, App::LOG_BACKEND
+  ::Log.setup "*", log_level, App::LOG_BACKEND
 end
```
Comment (caspiano) [inline src/config.cr]:
```suggestion
log_level = App.running_in_production? ? Log::Severity::Info : Log::Severity::Debug
::Log.setup("*", log_level, App::LOG_BACKEND)
```

File: src/models/auth.cr [hunk 3]
```diff
@@ -1,9 +1,17 @@
 class Auth
+  @signing_key : String
+
   def initialize(@scopes : String)
-    @file_path = File.expand_path(ENV["GOOGLE_AUTH_FILE"])
+    @signing_key = read_base64_key(ENV["GOOGLE_PRIVATE_KEY"])
+    @issuer = ENV["GOOGLE_CLIENT_EMAIL"]
```
Comment (caspiano) [inline src/models/auth.cr]: Potential for key error here.

File: src/config.cr (hunk 2 above)
Comment (caspiano) [reply src/config.cr]: Is `App::LOG_BACKEND` ours? [place-labs/log-backend](https://github.com/place-labs/log-backend)

File: shard.yml [hunk 4]
```diff
@@ -1,5 +1,5 @@
-name: spider-gazelle
-version: 1.0.0
+name: wallet_api
+version: 1.1.0
```
Comment (caspiano) [inline shard.yml:2]: Need a crystal version here.

File: src/config.cr (hunk 2 above)
Comment (dukenguyenxyz) [reply src/config.cr]: нет

File: src/models/auth.cr (hunk 3 above)
Comment (dukenguyenxyz) [reply src/models/auth.cr]: what do you mean, if we dont load env, won't it just throw a missing env anyways?

File: src/constants.cr [hunk 5]
```diff
@@ -19,6 +19,35 @@ module App
...
+  APPLE_DESIGN_LABEL_COLOR      = ENV["APPLE_DESIGN_LABEL_COLOR"]
+  APPLE_LOGO_PATH               = ENV["APPLE_LOGO_PATH"]
+  APPLE_ICON_PATH               = ENV["APPLE_ICON_PATH"]
+  APPLE_LOGO_DESCRIPTION        = ENV["APPLE_LOGO_DESCRIPTION"]
+
+  # AWS
+  AWS_REGION = ENV["AWS_REGION"]
+  AWS_KEY    = ENV["AWS_KEY"]
+  AWS_SECRET = ENV["AWS_SECRET"]
+  AWS_BUCKET = ENV["AWS_BUCKET"]
```
Comment (caspiano) [inline src/constants.cr]: A more friendly way of doing these strict lookups

```crystal
  def self.required_environment(key)
    ENV[key]?.presence || abort("missing required environment variable #{key}")
  end
```

File: src/constants.cr [hunk 6]
```diff
@@ -19,6 +19,35 @@ module App
...
+
   def self.running_in_production?
     ENVIRONMENT == "production"
   end
```
Comment (caspiano) [inline src/constants.cr]:
```suggestion
  class_getter? production = ENVIRONMENT.downcase == "production"
```

File: src/config.cr (hunk 2 above)
Comment (caspiano) [reply src/config.cr]: Can't help to include it. Ideally, we will have the same observability as the core platform

**Review by caspiano — APPROVED** [review]

**Pull request merged**